### PR TITLE
Addition of T/N workflow

### PR DESCRIPTION
**Pull request by reisingerf**

### CODE REVIEW

File: data_processors/pipeline/lambdas/tumor_normal.py
Comment (alexiswl) [inline data_processors/pipeline/lambdas/tumor_normal.py:125]: Do we have this input template ready to go? Can we tell if the event attributes `fastq_list_rows` and `tumor_fastq_list_rows` are json-strings or dicts/objects?

File: data_processors/pipeline/lambdas/orchestrator.py
Comment (alexiswl) [inline data_processors/pipeline/lambdas/orchestrator.py:151]: I'm not familiar / across the LabMetadata models but I assume this is where the magic is in terms of collecting a list of 'records' containing WGS records that can be mapped to the fastq list rows objects in the lines below?

File: data_processors/pipeline/lambdas/tumor_normal.py
Comment (reisingerf) [reply data_processors/pipeline/lambdas/tumor_normal.py:125]: The input template is stored in [SSM parameter store](https://ap-southeast-2.console.aws.amazon.com/systems-manager/parameters/?region=ap-southeast-2&tab=Table#list_parameter_filters=Name:Contains:workflow%2Ftumor) analogous to the other workflows.

The fastq_list_rows are JSON serialisable dicts (generated from the FastqListRow objects).

File: data_processors/pipeline/lambdas/orchestrator.py
Comment (reisingerf) [reply data_processors/pipeline/lambdas/orchestrator.py:151]: Yup. Here we first query for metadata records (WGS, tumor samples) of a single subject. (same for normals)
Then further down we try to find FastqListRow entries for those metadata records (based on the library ID).

Still needs to deal with topup/rerun cases, but if we have at least one tumor and one normal record/FastqListRow, we proceed to preparing the T/N job.

File: data_processors/pipeline/lambdas/orchestrator.py
Comment (victorskl) [inline data_processors/pipeline/lambdas/orchestrator.py:76]: @reisingerf Let us go through this routine again. On the second look, this may need more a bit more on filtering status. And, also we can enhance/improve into Service layer and WorkflowManager for query, etc. I will explain why so.

File: data_processors/pipeline/lambdas/tumor_normal.py
Comment (alexiswl) [reply data_processors/pipeline/lambdas/tumor_normal.py:125]: Yep, just thought I'd clarify given we have the germline under [main.tf](https://github.com/umccr/infrastructure/blob/5e08f2028dee2754da12ba92c92516b6dc06380c/terraform/stacks/umccr_data_portal/workflow/main.tf#L99-L121). I can add the somatic to this list too? Or are you happy for it just to be registered manually in ssm for now?

**Review by victorskl — COMMENTED** [review]
Minor syntax and changes to pass `make test` testing suite. And, might still need to work a bit more on Germline workflows running or completed query.

File: data_processors/pipeline/lambdas/tumor_normal.py
Comment (reisingerf) [reply data_processors/pipeline/lambdas/tumor_normal.py:125]: Ah, yes! Good point!
Please add there. I just put it in SSM manually and forgot about it.

File: data_processors/pipeline/lambdas/orchestrator.py
Comment (reisingerf) [reply data_processors/pipeline/lambdas/orchestrator.py:76]: Yup, that definitely needs some work.
Should take workflow reruns into account as well, I guess.

File: data_processors/pipeline/constant.py
Comment (victorskl) [reply data_processors/pipeline/constant.py:99]: @alexiswl what you think? ok to remove this field for good?

Comment (reisingerf) [reply data_processors/pipeline/constant.py:99]: There does not seem to be a corresponding TF entry.
And `engine_parameters` does not seem to have one either, but is definitely used. Perhaps we should add that to the TF stack?

Comment (victorskl) [reply data_processors/pipeline/constant.py:99]: Right, yep agree, `engine_parameters` should be better managed with Workflow TF sub-stack, too!

File: data_processors/pipeline/lambdas/germline.py
Comment (reisingerf) [reply data_processors/pipeline/lambdas/germline.py]: Resolved by de8aff75

File: data_processors/pipeline/lambdas/orchestrator.py
Comment (reisingerf) [reply data_processors/pipeline/lambdas/orchestrator.py]: Resolved by 05b1b3e

File: data_processors/pipeline/lambdas/orchestrator.py
Comment (reisingerf) [reply data_processors/pipeline/lambdas/orchestrator.py]: Resolved by 05b1b3e

File: data_processors/pipeline/lambdas/orchestrator.py
Comment (reisingerf) [reply data_processors/pipeline/lambdas/orchestrator.py]: Resolved by 05b1b3e

File: data_processors/pipeline/lambdas/orchestrator.py
Comment (reisingerf) [reply data_processors/pipeline/lambdas/orchestrator.py:308]: Resolved by 05b1b3e

File: data_processors/pipeline/constant.py
Comment (alexiswl) [reply data_processors/pipeline/constant.py:99]: Sure! I thought this might be used in tf but I think we've just put the settings in the portal for now.

File: data_processors/pipeline/lambdas/orchestrator.py
Comment (reisingerf) [reply data_processors/pipeline/lambdas/orchestrator.py]: Addressed in 8b7aa66

File: data_portal/models.py
Comment (victorskl) [inline data_portal/models.py:549]: Syntax, need remove `objects` like:

```
qs: QuerySet = self.filter(
```

File: data_portal/models.py
Comment (victorskl) [inline data_portal/models.py:557]: Here too:

```
qs: QuerySet = self.filter(
```

**Review by victorskl — APPROVED** [review]
LGTM impl-wise! After fixing syntax, this is good to go merging. Great stuff, Flo!

File: data_portal/models.py
Comment (reisingerf) [reply data_portal/models.py]: Fixed in 489f45e

File: data_portal/models.py
Comment (reisingerf) [reply data_portal/models.py]: Fixed in 489f45e

File: data_processors/pipeline/lambdas/tumor_normal.py
Comment (reisingerf) [reply data_processors/pipeline/lambdas/tumor_normal.py:125]: Added SSM parameters to TF in https://github.com/umccr/infrastructure/commit/4d77bc2585ec1bd2358107e05e6c431d581ee19f

File: data_processors/pipeline/constant.py
Comment (reisingerf) [reply data_processors/pipeline/constant.py]: Unused method removed in 98209f3

**Comment by reisingerf** [comment]
@alexiswl any final comments/requests or happy to merge?

**Pull request merged**
